core/blockchain/block_tree.hpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_BLOCK_TREE_HPP
+#define KAGOME_BLOCK_TREE_HPP
+
+#include <vector>
+
+#include <outcome/outcome.hpp>
+#include "primitives/block.hpp"
+#include "primitives/block_id.hpp"
+#include "primitives/justification.hpp"
+
+namespace kagome::blockchain {
+  /**
+   * Storage for blocks, which has a form of tree; it serves two functions:
+   *    - keep tracking of all finalized blocks (they are kept in the
+   *      non-volatile storage)
+   *    - work with blocks, which participate in the current round (handling
+   *      forks, pruning the blocks, resolving child-parent relations, etc)
+   */
+  struct BlockTree {
+    using BlockHashVecRes = outcome::result<std::vector<primitives::BlockHash>>;
+
+    virtual ~BlockTree() = default;
+
+    /**
+     * Get a body (extrinsics) of the block
+     * @param block - id of the block to get body for
+     * @return body, if the block exists in our storage, error otherwise
```

Comment:
Is it correct that if the block doesn't exist, there still may be a problem with our storage? Maybe rephrase it then?